test/gtest/uct/ib/test_sockaddr.cc

```diff
@@ -855,6 +839,214 @@ UCS_TEST_P(test_uct_cm_sockaddr, connect_client_to_server_reject_with_delay)
+    wait_for_client_server_counters(&m_server_connect_cb_cnt,
+                                    &m_client_connect_cb_cnt, m_clients_num);
+
+    EXPECT_EQ(m_clients_num, m_server_recv_req_cnt);
+    EXPECT_EQ(m_clients_num, m_client_connect_cb_cnt);
+    EXPECT_EQ(m_clients_num, m_server_connect_cb_cnt);
+    EXPECT_EQ(m_clients_num, (int)m_server->num_eps());
+
+    /* Disconnect */
+    srand(seed);
+    UCS_TEST_MESSAGE << "Using seed: " << seed;
+
+    pthread_mutex_init(&m_lock, NULL);
+    m_all_eps.resize(2 * m_clients_num);
+
+    /* save all the clients' and server's eps in the m_all_eps array */
+    for (i = 0; i < m_clients_num; ++i) {
+        /* first 2 entities are m_server and m_client */
+        m_all_eps[i].ep                    = m_entities.at(2 + i).ep(0);
+        m_all_eps[i].state                 = 0;
+        m_all_eps[m_clients_num + i].ep    = m_server->ep(i);
+        m_all_eps[m_clients_num + i].state = 0;
+    }
+
+    /* Disconnect */
+    /* go over the eps array and for each ep - use rand() to decide whether or
+     * not it should initiate a disconnect */
+    for (i = 0; i < (2 * m_clients_num); ++i) {
+        if (ucs::rand() % 2) {
+            /* don't start a disconnect on an ep that was already disconnected */
+            pthread_mutex_lock(&m_lock);
+            if (!(m_all_eps[i].state & TEST_EP_FLAG_DISCONNECT_CB_INVOKED)) {
+                m_all_eps[i].state |= TEST_EP_FLAG_DISCONNECT_INITIATOR;
+                ASSERT_UCS_OK(uct_ep_disconnect(m_all_eps[i].ep, 0));
+                /* count the number of eps that initiated a disconnect */
+                m_ep_init_disconnect_cnt++;
+            }
+            pthread_mutex_unlock(&m_lock);
+        }
+    }
+
+    /* wait for all the disconnect flows that began, to complete.
+     * if an ep initiated a disconnect, its disconnect callback should have been
+     * called, and so is the disconnect callback of its remote peer ep.
+     * every ep that initiated a disconnect is counted. this couneter is
+     * decremented in its disconnect cb, therefore once all eps that initiated
+     * a disconnect are disconnected, this counter should be equal to zero */
+    deadline = ucs_get_time() + ucs_time_from_sec(10 * DEFAULT_TIMEOUT_SEC) *
+                                ucs::test_time_multiplier();
+
+    while ((m_ep_init_disconnect_cnt != 0) && (ucs_get_time() < deadline)) {
+        progress();
+    }
+    EXPECT_EQ(0, m_ep_init_disconnect_cnt);
+
+    /* count and print the number of eps that were not disconnected */
+    for (i = 0; i < (2 * m_clients_num); i++) {
+        if (m_all_eps[i].state == 0) {
+            no_disconnect_eps_cnt++;
+        } else {
+            EXPECT_TRUE((m_all_eps[i].state ==
+                        (TEST_EP_FLAG_DISCONNECT_INITIATOR |
+                         TEST_EP_FLAG_DISCONNECT_CB_INVOKED)) ||
+                        (m_all_eps[i].state == TEST_EP_FLAG_DISCONNECT_CB_INVOKED));
```

Comment:
EXPECT_TRUE(m_all_eps[i].state & TEST_EP_FLAG_DISCONNECT_CB_INVOKED)

Reply:
but then other bits may be on as well (which would be wrong).

Reply:
Which other flags? i see only 2 flags defined for EP state

Reply:
there could be a bug and other bits might be set.
checking if one bit is set is "less" than checking for the exact value we're expecting. wdyt?

Reply:
ok. so this check could be written shorter
```
EXPECT_TRUE((state & ~INITIATOR) == CB_INVOKED)
```